### ISSUE TITLE
Create a new HU & BU only Invoice & Create a BU Internal Invoice CSV

### ISSUE THREAD

**Issue by joachimweyl** [CCI-MOC/process_csv_report#25]
## Motivation
Automate the creation of a CSV of the Invoice sheet that the BU billing team requires for creating internal invoicing. Also, create a CSV of only the institutions that are currently being billed (HU & BU).

## Completion Criteria
CSV output of the BUInternalInvoice sheet created at the same time as the invoice scripts. CSV output of only the currently being billed Invoice data, currently only BU & HU.

### BU & HU CSV
- [x] Create a CSV that filters on only BU & HU Addresses
- [ ] test that it matches the data in the original Invoice CSV

### BU Internal Invoice
- [x] Review the [BUInternalInvoice sheet](https://docs.google.com/spreadsheets/d/1ZZELpyIpmcmCCwZXt3wiccKke8ljyW_WFOyd1NnMIVY/edit?pli=1#gid=1763617817)
- [x] Column 1 is Invoice Month
- [x] Column 2 is Manager PI
- [x] Column 3 is Project, can pull from ColdFront or filter Project - Allocation to remove the Allocation, in google sheets that can be done with this code `=REGEXEXTRACT(D2, "(.+)-")`
- [ ] then remove duplicate Project rows
- [ ] Column 4 is Cost which is a sum of all Cost from the main CSV where Project - Allocation includes the string that is in column 3 or from coldfront just the rows that match that Project
- [x] Column 5 is Credit same as column 4 but for Credit
- [x] Column 6 is Subsidy this one is a bit complex you can look at the function in the sheet or you can follow these rules
   - Each PI receives up to $100 discount across all their projects
   - If they have less than $100 usage then whatever they used is subsidized
   - If they used more than $100 remove up to $100 from each row of their projects until $100 is removed
- [x] Column 7 is Balance = Cost - Credit - Subsidy

## Completion dates
Desired - 2024-04-25
Required - TBD

**Comment by joachimweyl** [comment]
Transferred issue to avoid duplication.

**Comment by QuanMPhm** [comment]
@joachimweyl For each BU PI, do we apply the Credit, or the BU Subsidy first? This won't impact the final balance for each BU PI, but it will show how much credit or subsidy they end up using. I.e if the total cost of projects of a new BU PI was $1050, we could show that the PI used $1000 of credit and $50 of subsidy, or $950 of credit and $100 of subsidy.

**Comment by joachimweyl** [comment]
Credits first

**Comment by joachimweyl** [comment]
I think we missed one of the steps "then remove duplicate Project rows" when generating the BU internal csv

**Comment by joachimweyl** [comment]
It will look like [this](https://docs.google.com/spreadsheets/d/1pZ7_QCTj5TPBDn4a_4bhF4f5OYeBN7TEQGuw5umtQXM/edit?pli=1#gid=1656492659). The InternalBUInvoice will have only one row for each project (not project - allocation). How I did this in excel was to remove allocation and then remove duplicates which leaves just one row for each project. I then summed the values from the combined sheet when they contained that project and PI which left a row with the total cost of that project (not each individual Allocation of that project).
